src/libstd/ffi/c_str.rs

```diff
@@ -266,11 +266,15 @@ impl CString {
 
     /// Transfers ownership of the string to a C caller.
     ///
-    /// The pointer must be returned to Rust and reconstituted using
-    /// `from_raw` to be properly deallocated. Specifically, one
+    /// Unless the same memory allocator is used for both Rust and C,
+    /// the acquired pointer must be returned to Rust and reconstituted
+    /// using `from_raw` to be properly deallocated. Specifically, one
     /// should *not* use the standard C `free` function to deallocate
     /// this string.
     ///
+    /// More details about custom memory allocators can be found in
+    /// [the book](../../../book/custom-allocators.html).
```

Comment:
So the tricky bit here is that this has changed on master, that is

* https://doc.rust-lang.org/book/custom-allocators.html exists
* https://doc.rust-lang.org/nightly/book/custom-allocators.html does not. 

What you really want is https://doc.rust-lang.org/nightly/unstable-book/allocator.html, and this page is at https://doc.rust-lang.org/stable/std/ffi/struct.CStr.html

so this should be

    ../../unstable-book/allocator.html

I believe.